### ISSUE TITLE
Some problem about interpolate

### ISSUE THREAD

**Issue by Luoxd1996** [wolny/pytorch-3dunet#3]
I have cloned your code to my PC, and i find a AttributeError which is "module 'torch.nn.functional' has no attribute interpolate" .In pytorch forums i find a similar question "https://discuss.pytorch.org/t/attributeerror-module-torch-nn-functional-has-no-attribute-instance-norm/15997/4".
my pytorch version is 0.4.0,GPU is GTX1080Ti.Thanks!
I have solved this problem by change this funfunction.
    def forward(self, encoder_features, x):
        if self.upsample is None:
            output_size = encoder_features.size()[2:]
            x = F.interpolate(x, size=output_size, mode='nearest')
        else:
            x = self.upsample(x)
        # concatenate encoder_features (encoder path) with the upsampled input
        x = torch.cat((encoder_features, x), dim=1)
        x = self.double_conv(x)
        return x
change this line code"x = F.interpolate(x, size=output_size, mode='nearest')" to "x = F.upsample(x, size=output_size, mode='nearest')"

**Comment by Luoxd1996** [comment]
def _images_from_batch(self, name, batch):
        tag_template = '{}/batch_{}/channel_{}/slice_{}'

        slice_idx = batch.shape[2] // 2  # get the middle slice
        tagged_images = []
        for batch_idx in range(batch.shape[0]):
            for channel_idx in range(batch.shape[1]):
                tag = tag_template.format(name, batch_idx, channel_idx,
                                          slice_idx)
                img = batch[batch_idx, channel_idx, slice_idx, ...]
                tagged_images.append((tag, (self._normalize_img(img))))
        return tagged_images
this function return a list in Trainer the length of list should be 2，but the len(tagged_images) is 1.So I can't understand how to let the code work.Thanks!

**Comment by Luoxd1996** [comment]
def _log_images(self, input, target, prediction):
        sources = {
            'inputs': input.data.cpu().numpy(),
            'targets': target.data.cpu().numpy(),
            'predictions': prediction.data.cpu().numpy()
        }
        for name, batch in sources.items():
            for tag, image in self._images_from_batch(name, batch):
                self.writer.add_image(tag, image, self.num_iterations)
I find this function has a bug. which this image is a numpy array not a tensor,so should add a line "image = torch.from_numpy(image)".

**Comment by wolny** [comment]
Hi @Luoxd1996, re interpolate issue: you're using outdated version of pytorch. Please use the newest 0.4.1. If you look at the code in pytorch you'll notice that `F.upsample` uses `F.interpolate` internally, so your change is unnecessary in 0.4.1.

The other function you've mentioned works perfectly fine. Which version of tensorboard/tensorboardx are you using? I've tested it with `tensorboard 1.9.0` and `tensorboardx 1.4`

**Comment by Luoxd1996** [comment]
I am very glad that you can reply to me soon.my pytorch version is 0.4.0,
and my tensorboard version is 1.8.0. the next I will update my pytorch and
tensordoard, Thank you very much.in addition, I also have a question which
is I can't work this code in some other 3D dataset like this(
http://medicaldecathlon.com/).most of questions are dims mismatch,So I want
to ask you if you test on some 3D dataset.Thank you very much.

Adrian Wolny <notifications@github.com> 于2018年10月1日周一 下午5:12写道：

**Comment by wolny** [comment]
yes I've been using the implementation with a 3D light sheet microscopy data that I'm working on currently (it's not open unfortunately). I've also tested it on CREMI https://cremi.org/data/. Thanks for point out the http://medicaldecathlon.com/. I might write an example of how to train on some real world 3D dataset to make it clearer of how to use the code.

**Comment by Luoxd1996** [comment]
wow! hat sounds great! I'm really looking forward to your tutorial.the
source of 3d-unet in pytorch is too little to find a complete project,some
of code even can't work! Thank your very much!

Adrian Wolny <notifications@github.com> 于2018年10月1日周一 下午8:58写道：
